Realm/RLMQueryUtil.mm

```diff
@@ -190,64 +201,80 @@ bool has_any_to_many_links() const {
         });
     }
 
-    util::Optional<ColumnReference> last_link_column() const {
-        if (!m_links.size()) {
-            return util::none;
-        }
-        auto links = m_links;
-        RLMProperty *property = links.back();
-        links.resize(links.size() - 1);
-        return ColumnReference(m_schema, property, std::move(links));
+    ColumnReference last_link_column() const {
+        REALM_ASSERT(!m_links.empty());
+        return {*m_query, *m_group, m_schema, m_links.back(), {m_links.begin(), m_links.end() - 1}};
     }
 
-    ColumnReference column_ignoring_links() const {
-        return {m_schema, m_property};
+    ColumnReference column_ignoring_links(Query& query) const {
+        return {query, *m_group, m_schema, m_property};
     }
 
 private:
-    Table* table_for_query(Query& query) const
+    template <typename T, typename... SubQuery>
+    auto resolve_backlink(std::true_type, SubQuery&&... subquery) const
```

Comment:
I think I'd prefer something closer to your original approach until we're able to use `if constexpr` directly.